game/scripts/vscripts/TutorialGraph/Steps.ts

```diff
@@ -116,6 +116,29 @@ export const spawnUnit = (unitName: tg.StepArgument<string>, spawnLocation: tg.S
     })
 }
 
+/**
+ * Creates a tutorial step that returns a fork of multiple moveUnit steps.
+ * @param unit The unit to move.
+ * @param moveLocations Array of locations the unit should move through.
+ * @param completeIfUnitInvalid Optional param that controls whether steps should complete if provided unit is invalid or dead. Default value is false
+ */
+export const moveUnitSequence = (unit: tg.StepArgument<CDOTA_BaseNPC>, moveLocations: tg.StepArgument<Vector[]>, completeIfUnitInvalid: boolean = false): tg.TutorialStep => {
+
+    const steps = (context: tg.TutorialContext) => {
+        const moveUnitSteps: tg.TutorialStep[] = []
+        const actualUnit = tg.getArg(unit, context)
+        const actualMoveLocations = tg.getArg(moveLocations, context)
+
+        for (const moveLocation of actualMoveLocations) {
+            moveUnitSteps.push(moveUnit(actualUnit, moveLocation, completeIfUnitInvalid))
+        }
+
+        return moveUnitSteps
+    }
+
+    return tg.fork(steps)
```

Comment:
Shouldn't this be a `seq`? Otherwise it would do all the moves in parallel, but judging from the name I assume it should be sequentially?

Reply:
I did it in parallel because I expected the commands to be queued up, and thought it would result in a smoother movement since there's an idle check in the moveUnit step. I didn't actually try using seq to see the result, maybe it's ok.

Reply:
Works as long as the queueing always happens and fork executes them in the given order I guess.